From 57acfbe5bbc1662439c7acf52a727e947d71622e Mon Sep 17 00:00:00 2001
From: Maxime Ripard <mripard@kernel.org>
Date: Fri, 19 Jun 2026 14:25:19 +0200
Subject: [PATCH 00687/15333] drm/atomic-state-helper: Remove
 drm_atomic_helper_bridge_reset()

All drivers have been converted to the atomic_create_state callback
and its drm_atomic_helper_bridge_create_state() helper.

Remove the deprecated drm_atomic_helper_bridge_reset().

Reviewed-by: Thomas Zimmermann <tzimmermann@suse.de>
Reviewed-by: Luca Ceresoli <luca.ceresoli@bootlin.com>
Tested-by: Luca Ceresoli <luca.ceresoli@bootlin.com> # imx8mp + sn65dsi84 + bridge hotplug
Link: https://patch.msgid.link/20260619-drm-no-more-bridge-reset-v3-74-ff399263111b@kernel.org
Signed-off-by: Maxime Ripard <mripard@kernel.org>
---
 drivers/gpu/drm/drm_atomic_state_helper.c | 19 +------------------
 include/drm/drm_atomic_state_helper.h     |  2 --
 2 files changed, 1 insertion(+), 20 deletions(-)

diff --git a/drivers/gpu/drm/drm_atomic_state_helper.c b/drivers/gpu/drm/drm_atomic_state_helper.c
index cfb54fc853ef..fbdc3d893a0c 100644
--- a/drivers/gpu/drm/drm_atomic_state_helper.c
+++ b/drivers/gpu/drm/drm_atomic_state_helper.c
@@ -873,8 +873,7 @@ EXPORT_SYMBOL(drm_atomic_helper_bridge_duplicate_state);
  * @state: bridge state to destroy
  *
  * Destroys a bridge state previously created by
- * &drm_atomic_helper_bridge_create_state(),
- * &drm_atomic_helper_bridge_reset() or
+ * &drm_atomic_helper_bridge_create_state() or
  * &drm_atomic_helper_bridge_duplicate_state(). This helper is meant to be
  * used as a bridge &drm_bridge_funcs.atomic_destroy_state hook for bridges
  * that don't subclass the bridge state.
@@ -907,22 +906,6 @@ void __drm_atomic_helper_bridge_state_init(struct drm_bridge_state *state,
 }
 EXPORT_SYMBOL(__drm_atomic_helper_bridge_state_init);
 
-/**
- * drm_atomic_helper_bridge_reset() - Allocate and initialize a bridge state
- *				      to its default
- * @bridge: the bridge this state refers to
- *
- * Allocates the bridge state and initializes it to default values. This helper
- * is meant to be used as a bridge &drm_bridge_funcs.atomic_reset hook for
- * bridges that don't subclass the bridge state.
- */
-struct drm_bridge_state *
-drm_atomic_helper_bridge_reset(struct drm_bridge *bridge)
-{
-	return drm_atomic_helper_bridge_create_state(bridge);
-}
-EXPORT_SYMBOL(drm_atomic_helper_bridge_reset);
-
 /**
  * drm_atomic_helper_bridge_create_state - default
  *              &drm_bridge_funcs.atomic_create_state hook for bridges
diff --git a/include/drm/drm_atomic_state_helper.h b/include/drm/drm_atomic_state_helper.h
index 68c685ad330f..34a599c3d86d 100644
--- a/include/drm/drm_atomic_state_helper.h
+++ b/include/drm/drm_atomic_state_helper.h
@@ -106,5 +106,3 @@ void __drm_atomic_helper_bridge_state_init(struct drm_bridge_state *state,
 					   struct drm_bridge *bridge);
 struct drm_bridge_state *
 drm_atomic_helper_bridge_create_state(struct drm_bridge *bridge);
-struct drm_bridge_state *
-drm_atomic_helper_bridge_reset(struct drm_bridge *bridge);
-- 
2.55.0

